### PR TITLE
feat: Create GraalVM Native/ Native-17 Jobs

### PR DESCRIPTION
**Pull request by lqiu96**
Testing of GraalVM IT Jobs done at https://github.com/googleapis/google-cloud-java/pull/8148

### CODE REVIEW

File: templates/.kokoro/build.sh [hunk 1]
```diff
@@ -64,8 +63,11 @@ if [[ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" && "${GOOGLE_APPLICATION_CREDENTI
   export GOOGLE_APPLICATION_CREDENTIALS=$(realpath ${KOKORO_GFILE_DIR}/${GOOGLE_APPLICATION_CREDENTIALS})
 fi
 
+if [ -f "${KOKORO_GFILE_DIR}/secret_manager/java-bigqueryconnection-samples-secrets" ]; then
+  source "${KOKORO_GFILE_DIR}/secret_manager/java-bigqueryconnection-samples-secrets"
+fi
```
Comment (lqiu96) [inline templates/.kokoro/build.sh:68]: GraalVM IT jobs also require this secret

File: templates/.kokoro/build.sh [hunk 2]
```diff
@@ -81,14 +83,55 @@ case ${JOB_TYPE} in
...
+    # grep returns 1 (error code) and exits the pipeline if there is no match
+    # If there is no match, it will return true so the rest of the commands can run
+    modified_files=$(git diff --name-only $KOKORO_GITHUB_PULL_REQUEST_COMMIT $KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH)
+    printf "Modified files:\n%s\n" "${modified_files}"
+
+    # If root pom.xml is touched, run ITs on all the modules
+    root_pom_modified=$(echo "${modified_files}" | grep -e '^pom.xml$' || true)
+    if [[ -n $root_pom_modified ]]; then
+      TEST_ALL=true
+      echo "Testing the entire monorepo"
```
Comment (lqiu96) [inline templates/.kokoro/build.sh:98]: If root pom.xml is modified, the IT for entire monorepo is run

File: templates/.kokoro/build.sh [hunk 3]
```diff
@@ -23,38 +23,37 @@ cd ${scriptDir}/..
...
 cp settings.xml ${HOME}/.m2
 
+function assign_modules_to_job() {
+  modules=$(mvn help:evaluate -Dexpression=project.modules | grep '<.*>.*</.*>' | sed -e 's/<.*>\(.*\)<\/.*>/\1/g')
+  module_list=()
+  num=0
+  for module in $modules
+  do
+    # Add 1 as JOB_NUMBER is 1-indexed instead of 0-indexed
+    mod_num=$((num % NUM_JOBS + 1))
```
Comment (lqiu96) [inline templates/.kokoro/build.sh:37]: NUM_JOB is ENV_VAR passed in. We can create new .cfg files and update this value in the common build.proto

File: templates/.kokoro/presubmit/graalvm-sub-jobs/native-17/common.cfg [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Format: //devtools/kokoro/config/proto/build.proto
+
+# Build configs are duplicated in native/ and native-17/
+# Build configs don't inherit the hierarchical common.cfg: qaqs/6158654898243108864
```
Comment (lqiu96) [inline templates/.kokoro/presubmit/graalvm-sub-jobs/native-17/common.cfg:4]: FYI, job.proto inherits hierarchical common.cfg but build.proto doesn't. Another option is to have one common.cfg in presubmit/ and have all 40 .cfg files in presubmit/ so we don't have to maintain two common.cfg

**Comment by lqiu96** [comment]
Waiting for the latest build to run -- Made some cleanup changes and updates to the configs

**Comment by lqiu96** [comment]
@suztomo @meltsufin The jobs worked with the latest updates. Can I get a review? Thanks!

File: templates/.kokoro/build.sh [hunk 5]
```diff
@@ -81,14 +83,55 @@ case ${JOB_TYPE} in
...
   integration)
-    if [ -f "${KOKORO_GFILE_DIR}/secret_manager/java-bigqueryconnection-samples-secrets" ]
-           then
-               source "${KOKORO_GFILE_DIR}/secret_manager/java-bigqueryconnection-samples-secrets"
+    TEST_ALL=false
+    # Find the files changed from when the PR branched to the last commit
+    # Filter for java modules and get all the unique elements
+    # grep returns 1 (error code) and exits the pipeline if there is no match
+    # If there is no match, it will return true so the rest of the commands can run
+    modified_files=$(git diff --name-only $KOKORO_GITHUB_PULL_REQUEST_COMMIT $KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH)
```
Comment (suztomo) [inline templates/.kokoro/build.sh:91]: The modified files moved to here. Nice.

**Review by suztomo — APPROVED** [review]

**Pull request merged**